clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroComputeRequestBuilder.java

```diff
@@ -379,7 +381,18 @@ private void streamingExecuteInternal(
     executed = true;
 
     long preRequestTimeInNS = time.nanoseconds();
-    SchemaAndToString resultSchema = getResultSchema();
+    SchemaAndToString resultSchema;
+    try {
+      resultSchema = getResultSchema();
+    } catch (VeniceClientException e) {
+      // If the schema is invalid, we need to refresh the schema
+      this.latestValueSchemaId = schemaReader.getLatestValueSchemaId();
```

Comment:
We should limit how often we do this since `VeniceClientException`'s scope is quite large. We could be triggering refreshes when the underlying problem cannot be fixed by schema refresh.